lib/brains/forms.py

```diff
@@ -110,3 +128,11 @@ def braintree_data(self):
             self.cleaned_data['bt_signature'],
             self.cleaned_data['bt_payload']
         )
+
+    def clean(self):
+        res = requests.post(settings.BRAINTREE_PROXY + '/parse', self.data)
+        if res.status_code != 204:
+            log.error('Did not receive a 204 from solitude-auth, got: {}'
+                      .format(res.status_code))
+            raise forms.ValidationError('Did not pass verification')
```

Comment:
but you'd have to protect against empty input, I suppose (which will cause a decode error before raising a 'field is required' error, maybe)